00xx-account-deletion.md

```diff
@@ -0,0 +1,471 @@
+[add_content](https://github.com/MinaProtocol/mina/blob/4495af5caea5e1bb2f98f92592c065f93a586ade/src/lib/syncable_ledger/syncable_ledger.ml#L408)
+needs to be handle this correctly and this in turn relies on
+[set_all_accounts_rooted_at_exn](https://github.com/MinaProtocol/mina/blob/4495af5caea5e1bb2f98f92592c065f93a586ade/src/lib/syncable_ledger/syncable_ledger.ml#L419).
+The change here will mainly be to call ``set_all_accounts_rooted_at_exn` as many
+times as there are chunks in the data.
+
+
+
+### Transaction logic and snark
+
+The support for deletion at the ledger level now needs to be lifted within the
+transaction logic and snark for account updates.
+The key idea here is to handle removal as a specific kind of account update for
+a given account.
+
+Concretely, the basic support is provided by adding a deletion flag on account
+updates, i.e., extend the `Account_update.Body.Stable.t` type with an additional
+record field `delete_account`.
+
+This turns the current definition into
+```ocaml
+      (* This will update the version of the correspond Mina_wire_types definition too).
+      type t = Mina_wire_types.Mina_base.Account_update.Body.V2.t =
+        { public_key : Public_key.Compressed.Stable.V1.t
+        ; token_id : Token_id.Stable.V2.t
+        ; update : Update.Stable.V1.t
+        ; balance_change :
+            (Amount.Stable.V1.t, Sgn.Stable.V1.t) Signed_poly.Stable.V1.t
+        ; increment_nonce : bool
+        ; events : Events'.Stable.V1.t
+        ; actions : Events'.Stable.V1.t
+        ; call_data : Pickles.Backend.Tick.Field.Stable.V1.t
+        ; preconditions : Preconditions.Stable.V1.t
+        ; use_full_commitment : bool
+        ; implicit_account_creation_fee : bool
+        ; may_use_token : May_use_token.Stable.V1.t
+        ; authorization_kind : Authorization_kind.Stable.V1.t
+         (* if true, this account update will delete the account *)
+        ; delete_account: bool;
+        }
+```
+
+This extension therefore provides a version 2 of the current versioned type (`V2`).
+
+This simple change trickles down into updating the transitive closure of all
+type definitions depending on `Account_update`, including
+`Mina_wire_types.Mina_transaction`.
+
+#### Effect of deleting an account
+
+Deleting an account $a₀$ results in two actions:
+1. Actual removal of account $a₀$ of the storage layer;
+2. Return of balance of MINA tokens from $a₉$ and initial creation fee to a specified $a₁$ account.
+
+Handling other tokens is the responsability of the smart contract(s) dealing with these tokens.
+
+Furthermore, account deletion should not be permitted except for some authorized
+participants, e.g., the contract that created the consumable account.
+
+We propose to restrict this capacity to the account to be deleted, so that only
+yourself can call for your own deletion.
+
+For that, we need to check this with respect to the `permissions` field of
+`Account_update.Update.t`.
```

Comment:
This record declares an update to the permissions, not the permissions themselves.